rediscluster/pipeline.py

```diff
@@ -142,144 +146,70 @@ def send_cluster_commands(self, stack, raise_on_error=True, allow_redirections=T
-                    # if we are stuck in a retry loop, slow things down a bit to give the failover
-                    # a chance of actually happening.
-                    if ttl < self.RedisClusterRequestTTL / 2:
-                        time.sleep(0.1)
-                    continue
-
-                # If cluster is down it should be raised and bubble up to
-                # utils.clusterdown_wrapper()
-                if isinstance(c.result, ClusterDownError):
-                    self.connection_pool.disconnect()
-                    self.connection_pool.reset()
-                    self.refresh_table_asap = True
-
-                    raise c.result
-
-                # A MOVED response from the cluster means that somehow we were misinformed about which node
-                # a given key slot maps to. This can happen during cluster resharding, during master-slave
-                # failover, or if we got a connection error and were forced to re-attempt the command against a
-                # random node.
-                if isinstance(c.result, MovedError):
-                    # Do not perform full cluster refresh on every MOVED error
-                    self.reinitialize_counter += 1
-
-                    if self.reinitialize_counter % self.reinitialize_steps == 0:
-                        self.refresh_table_asap = True
-
-                    node = self.connection_pool.nodes.set_node(c.result.host, c.result.port, server_type='master')
-                    self.connection_pool.nodes.slots[c.result.slot_id][0] = node
-                    self._fail_on_redirect(allow_redirections)
-
-                # an ASK error from the server means that only this specific command needs to be tried against
-                # a different server (not every key in the slot). This happens only during cluster re-sharding
-                # and is a major pain in the ass. For it to work correctly, we have to resend the command to
-                # the new node but only after first sending an ASKING command immediately beforehand.
-                elif isinstance(c.result, AskError):
-                    node = self.connection_pool.nodes.set_node(c.result.host, c.result.port, server_type='master')
-                    c.node = node
-                    c.asking = True
-                    self._fail_on_redirect(allow_redirections)
-
-                # if it is any other type of exception, retrying won't help.
-                # quit while you are ahead.
-                elif isinstance(c.result, Exception):
-                    break
-
-        # We are done trying. did we get stuck on connection or timeout errors?
-        if [c for c in attempt if isinstance(c.result, (ConnectionError, TimeoutError))] and ttl < 1:
-            raise ClusterError('TTL exhausted.')
-
-        # YAY! we made it out of the attempt loop.
+                try:
+                    c.result = super(StrictClusterPipeline, self).execute_command(*c.args, **c.options)
```

Comment:
Can you add some comment like `This is where the magic ASK/MOVED redirection magic happens inside the execute_command that is piggybacking on the regular redirection logic from the normal client class` It was unclear to me at first where it was moved, but with a hint it is here it will make it much clearer.